cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Insert.hs

```diff
@@ -296,14 +297,23 @@ insertPoolRegister tracer network (EpochNo epoch) txId idx params = do
         ]
 
   poolHashId <- insertPoolHash (Shelley._poolId params)
+
+  let wasPoolRegistered = Set.member (Shelley._poolId params) $ getPoolParams $ lssOldState lStateSnap
```

Comment:
I think this is big enough to be pulled out into a separate function, even if its in a `where` clause.